pstress/pstress-run.sh

```diff
@@ -1337,25 +1370,62 @@ elif [ "${VERSION_INFO}" != "5.7" -a "${VERSION_INFO}" != "8.0" ]; then
   echo "WARNING: mysqld (${BIN}) version detection failed. This is likely caused by using this script with a non-supported distribution or version of mysqld. Please expand this script to handle (which shoud be easy to do). Even so, the script will now try and continue as-is, but this may fail."
 fi
 
-# Keyring component is unsupported in 5.7
-  if [ "${VERSION_INFO}" == "5.7" ]; then
-    KEYRING_COMPONENT=0
+if [ ${ENCRYPTION_RUN} -eq 1 ]; then
+  if [ ${PXC} -eq 1 ]; then
+    if [ ${KEYRING_VAULT} -eq 0 -a ${KEYRING_FILE} -eq 0 ]; then
+      echoit "Enable atleast one encryption type (keyring_vault or keyring_file) if ENCRYPTION_RUN=1"
+      exit 1
+    fi
+  elif [ ${KEYRING_VAULT} -eq 0 -a ${KEYRING_FILE} -eq 0 -a ${KEYRING_COMPONENT} -eq 0 ]; then
+    echoit "Enable atleast one encryption type (keyring_vault or keyring_file or keyring_component) if ENCRYPTION_RUN=1"
+    exit 1
+  fi
+fi
+
+# Start vault server for pstress encryption run
+if [[ ${KEYRING_VAULT} -eq 1 ]];then
+  echoit "Setting up vault server"
+  mkdir $WORKDIR/vault
+  rm -rf $WORKDIR/vault/*
+  killall vault > /dev/null 2>&1
+  if [[ $PXC -eq 1 ]];then
+    ${SCRIPT_PWD}/vault_test_setup.sh --workdir=$WORKDIR/vault --setup-pxc-mount-points --use-ssl > /dev/null 2>&1
+  else
+    ${SCRIPT_PWD}/vault_test_setup.sh --workdir=$WORKDIR/vault --use-ssl > /dev/null 2>&1
+    VAULT_PARAM="--early-plugin-load=keyring_vault.so --keyring_vault_config=$WORKDIR/vault/keyring_vault_ps.cnf"
+  fi
+fi
+
+if [ ${KEYRING_FILE} -eq 1 ]; then
```

Comment:
please fix if you think its better to make this an elif, because it could be part of same code block.

Reply:
Sure, I will fix this.